website/docs/basics/blocks-styles.md

```diff
@@ -510,3 +561,229 @@ Attribute value replacement variable is used to return the attribute value where
+_Note: If there is a need for an extra variable, or overriding some of the automatically generated variables (from the helper). The variables will be outputted below the responsive variables. *Example2*_
+_Note2: It is important to take care of the order placed in responsive attributes since generating variables is relying on it. Also, ensure that you use `inverse` property in the right way._
+
+**Example1**
+```json
+{
+	"componentName": "wrapper",
+	"title": "Wrapper",
+	"componentClass": "wrapper",
+	"attributes": {
+		"wrapperHideLarge": {
+			"type": "boolean",
+			"default": false
+		},
+		"wrapperHideDesktop": {
+			"type": "boolean"
+		},
+		"wrapperHideTablet": {
+			"type": "boolean"
+		}
+	},
+	"responsiveAttributes": {
+		"wrapperHide": {
+			"large": "wrapperHideLarge",
+			"desktop": "wrapperHideDesktop",
+			"tablet": "wrapperHideTablet"
+		}
+	},
+	"variables": {
+		"wrapperHide": {
+			"true": [
+				{
+					"inverse": true,
+					"variable": {
+						"wrapper-display": "none"
+					}
+				}
+			],
+			"false": [
+				{
+					"inverse": true,
+					"variable": {
+						"wrapper-display": "var(--wrapper-display-type, grid)"
+					}
+				}
+			]
+		}
+	}
+}
+```
+
+**Transformed:**
+```json
+{
+	"componentName": "wrapper",
+	"title": "Wrapper",
+	"componentClass": "wrapper",
+	"attributes": {
+		"wrapperHide": {
+			"type": "boolean",
+			"default": false
+		}
+	},
+	"responsiveAttributes": {
+		"wrapperHideLarge": {
+			"type": "boolean",
+			"default": false
+		},
+		"wrapperHideDesktop": {
+			"type": "boolean"
+		},
+		"wrapperHideTablet": {
+			"type": "boolean"
+		}
+	},
+	"variables": {
+		"wrapperHideLarge": {
+			"true": [
+				{
+					"variable": {
+						"wrapper-display": "none"
+					}
+				}
+			],
+			"false": [
+				{
+					"variable": {
+						"wrapper-display": "var(--wrapper-display-type, grid)"
+					}
+				}
+			]
+		},
+		"wrapperHideDesktop": {
+			"true": [
+				{
+					"inverse": true,
+					"breakpoint": "desktop",
+					"variable": {
+						"wrapper-display": "none"
+					}
+				}
+			],
+			"false": [
+				{
+					"inverse": true,
+					"breakpoint": "desktop",
+					"variable": {
+						"wrapper-display": "var(--wrapper-display-type, grid)"
+					}
+				}
+			]
+		},
+		"wrapperHideTablet": {
+			"true": [
+				{
+					"inverse": true,
+					"breakpoint": "desktop",
+					"variable": {
+						"wrapper-display": "none"
+					}
+				}
+			],
+			"false": [
+				{
+					"inverse": true,
+					"breakpoint": "desktop",
+					"variable": {
+						"wrapper-display": "var(--wrapper-display-type, grid)"
+					}
+				}
+			]
+		}
+	}
+}
+```
+
+**Output**
+```css
+.wrapper[data-id='210c9bbf733ef5c6aa74c49168ac29a7'] {
+	--wrapper-display: var(--wrapper-display-type, grid);
```

Comment:
what is this --wrapper-display-type, grid thing? Where is it coming from and what does it do?

Reply:
I was not sure whether to use a real example for the wrapper hide option or to think of a new example. In the end, I decided to just copy from the real wrapper manifest. This `wrapper-display-type` is not important here at all. The whole purpose is of this code is not to show how variables work and where do they come from, but how will responsive variables behave.

Should I change it to just `grid` to make it clearer?

Reply:
Is this example something that comes out of the box in the wrapper?

The thing with examples is that somebody will copy/paste them to try them out so if you put something that doesn't exist it will cause an error and it won't be clear what was the intention of the example.

Reply:
This will not work for sure. You need JS(options), CSS, and PHP parts to get the full result.
This is showing an upgrade and if you follow the example, and layout your "problem" the same way as in the example, it will work completely fine.
`--wrapper-display-type` this part doesn't play any role because it will anyway do a fallback on `grid` so we are safe in the matter of unused variable.

Maybe I can add a link to the wrapper in our frontend libs? (even though we still didn't test the wrapper in its full capacity so we are not sure whether it will or will not be there as it is currently written on Infinum web)